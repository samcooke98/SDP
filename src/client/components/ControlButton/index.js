import React from "react";
import styles from "./web.css";

export default class ControlButton extends React.Component {
    render() {
        console.log(this.props);
        return (
            <div
                onMouseDown={(evt) => {
                    evt.preventDefault();
                    this.props.onClick(evt); 
                }
                }
                className={styles.controlBtn + " " + ( this.props.active ? styles.active : "")}
            >
                {this.props.children}
            </div>
        )

    }

}
