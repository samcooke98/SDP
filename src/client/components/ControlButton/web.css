
.controlBtn { 
    width:100px;
    height:100px;
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 0;
    padding: 0;

    width: 100px;
    height: 100px;
    vertical-align: middle;
    background-color: #E6E9ED;
    box-sizing: border-box;
    cursor: pointer;
}

.controlBtn * { 
    height: 24px;
    width: 24px; 
}

.active { 
    background-color: #8a8b8e;
}