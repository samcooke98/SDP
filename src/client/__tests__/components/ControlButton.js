import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import ControlButton from "../../components/ControlButton/index";


describe('React Component: ControlButton', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <ControlButton />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})