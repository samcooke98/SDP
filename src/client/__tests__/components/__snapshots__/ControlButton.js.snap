// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: ControlButton matches a snapshot 1`] = `
<div
  className="undefined "
  onMouseDown={[Function]}
  style={undefined}
/>
`;
